### PR TITLE
feat: enable sorting of notifications

### PR DESCRIPTION
**Pull request by mareklibra**
## Hey, I just made a Pull Request!

Notifications can be sorted by predefined list of options.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by mareklibra** [comment]
Screenshot from 2024-03-01 13-27-49

---

Screenshot from 2024-03-01 13-27-53

**Review by freben — COMMENTED** [review]
Looks good overall!

File: plugins/notifications/src/components/NotificationsFilters/NotificationsFilters.tsx [hunk 1]
```diff
@@ -24,24 +24,20 @@ import {
...
-  // };
-  // setSorting: ({
-  //   orderBy,
-  //   orderByDirec,
-  // }: {
-  //   orderBy: GetNotificationsOrderByEnum;
-  //   orderByDirec: GetNotificationsOrderByDirecEnum;
-  // }) => void;
+  sorting: SortBy;
+  setSorting: (sortBy: SortBy) => void;
```
[the remark this answers is not in the document]
Comment (mareklibra) [reply plugins/notifications/src/components/NotificationsFilters/NotificationsFilters.tsx:40]: Thanks, I will allign

[the remark this answers is not in the document]
Comment (mareklibra) [reply plugins/notifications/src/components/NotificationsFilters/NotificationsFilters.tsx]: done

File: plugins/notifications-backend/src/service/router.ts [hunk 2]
```diff
@@ -187,6 +209,12 @@ export async function createRouter(
     if (req.query.limit) {
       opts.limit = Number.parseInt(req.query.limit.toString(), 10);
     }
+    if (req.query.sort) {
+      opts.sort = getSort(req.query.sort.toString());
+    }
+    if (req.query.sort_order) {
```
[the remark this answers is not in the document]
Comment (mareklibra) [reply plugins/notifications-backend/src/service/router.ts:215]: Follow-up: https://github.com/backstage/backstage/pull/23450

**Pull request merged**